src/cdk/dialog/dialog-config.ts

```diff
@@ -132,6 +132,12 @@ export class DialogConfig<D = unknown, R = unknown, C extends BasePortalOutlet =
    */
   closeOnDestroy?: boolean = true;
 
+  /**
+   * Whether the dialog should close when the underlying overlay is disposed. This is useful if
+   * another service is wrapping the dialog and is managing the destruction instead.
+   */
+  closeOnOverlayDetachments?: boolean = true;
```

Comment:
I really don't like this pattern, but I had to do it, because the Material dialog needs to be able to control its own destruction in order to sync up its animation, and at the same time I didn't want to have special Material logic in the CDK implementation.

Reply:
Seems fine to me. One question: If you are addressing other stuff, could you maybe add a small example to the JSDoc to mention e.g. the scroll strategies that might cause the "detach"?